core/src/main/java/cucumber/runtime/arquillian/ArquillianCucumber.java

```diff
@@ -120,223 +127,108 @@ protected void runChild(final FrameworkMethod method, final RunNotifier notifier
-                        reporter.result(result);
-                    }
-
-                    @Override
-                    public void after(final Match match, final Result result) {
-                        reporter.after(match, result);
-                    }
-
-                    @Override
-                    public void embedding(final String mimeType, final byte[] data) {
-                        reporter.embedding(mimeType, data);
-                    }
-
-                    @Override
-                    public void write(final String text) {
-                        reporter.write(text);
-                    }
-                }, i18n);
-                EventHelper.fire(new AfterStep(featurePath, step));
-            }
+        final InputStream gluesInputStream = threadContextClassLoader.getResourceAsStream(ClientServerFiles.GLUES_LIST);
+        final Collection<Class<?>> glues = loadGlues(gluesInputStream, threadContextClassLoader, javaTestClass);
+        
+        final CucumberRuntime cucumberRuntime = new CucumberRuntime(null, threadContextClassLoader, Arrays.asList(new ArquillianBackend(glues, javaTestClass, testInstance)), runtimeOptions);
+        final Formatter formatter = runtimeOptions.formatter(threadContextClassLoader);
+        final JUnitReporter jUnitReporter = new JUnitReporter(runtimeOptions.reporter(threadContextClassLoader), formatter, runtimeOptions.isStrict());
+        runFeatures(cucumberFeatures, cucumberRuntime, jUnitReporter, runNotifier);
+        
+        if (reported) 
+        {
+            final String path = cukespaceConfigurationProperties.getProperty(CucumberConfiguration.REPORTABLE_PATH);
+            addReportTestIntoFile(path, javaTestClass, reportBuilder);
+        }
 
-            @Override
-            public void runBeforeHooks(final Reporter reporter, final Set<Tag> tags) {
-                EventHelper.fire(new BeforeBeforeHooks());
-                super.runBeforeHooks(reporter, tags);
-                EventHelper.fire(new AfterBeforeHooks());
+        handleCucumberTestErrors(cucumberRuntime.getErrors(), cucumberRuntime);
+        
+    }
+    
+    private static Properties loadCucumberConfigurationProperties(final InputStream configurationInputStream) throws Exception
+    {
+    	if (configurationInputStream != null) {
+    		return loadConfigurationPropertiesFromStream(configurationInputStream);
+        }
+    	else
+    	{
+    		return loadConfigurationPropertiesFromObject(CucumberConfiguration.instance());
+    	}
+    }
+    
+    private static Properties loadConfigurationPropertiesFromStream(final InputStream configurationInputStream) throws Exception
+    {
+    	Properties configurationProperties = new Properties();
+        configurationProperties.load(configurationInputStream);
+        return configurationProperties;
+    }
+    
+    private static Properties loadConfigurationPropertiesFromObject(final CucumberConfiguration cucumberConfiguration) throws Exception
```

Comment:
If you split it here maybe move it to configuration object itself as an utility method?